src/Core/Utils.fs

```diff
@@ -56,8 +56,11 @@ module Option =
 
 [<RequireQualifiedAccess>]
 module Document =
-    let (|FSharp|Other|) (document : TextDocument) =
-        if document.languageId = "fsharp" then FSharp else Other
+    let (|FSharp|CSharp|VB|Other|) (document : TextDocument) =
```

Comment:
Expanded this to include other msbuild-based languages, but it may be worth splitting msbuild out into a language-agnostic component?